src/types.d.ts

```diff
@@ -338,19 +338,29 @@ export interface ColumnData {
 /**
  * Parquet query options for reading data
  */
-export interface ParquetReadOptions {
+export type ParquetReadOptions =  ParquetArrayRowReadOptions | ParquetObjectRowReadOptions
+
+interface ParquetBaseReadOptions {
   file: AsyncBuffer // file-like object containing parquet data
   metadata?: FileMetaData // parquet metadata, will be parsed if not provided
   columns?: string[] // columns to read, all columns if undefined
-  rowFormat?: string // format of each row passed to the onComplete function
   rowStart?: number // first requested row index (inclusive)
   rowEnd?: number // last requested row index (exclusive)
   onChunk?: (chunk: ColumnData) => void // called when a column chunk is parsed. chunks may be outside the requested range.
-  onComplete?: (rows: any[][]) => void // called when all requested rows and columns are parsed
   compressors?: Compressors // custom decompressors
   utf8?: boolean // decode byte arrays as utf8 strings (default true)
 }
 
+interface ParquetArrayRowReadOptions extends ParquetBaseReadOptions {
+  rowFormat?: "array" // format of each row passed to the onComplete function
+  onComplete?: (rows: any[][]) => void // called when all requested rows and columns are parsed
+}
+
+interface ParquetObjectRowReadOptions extends ParquetBaseReadOptions {
+  rowFormat: "object" // format of each row passed to the onComplete function
+  onComplete?: (rows: any[]) => void // called when all requested rows and columns are parsed
```

Comment:
This `any` is too broad, should be `Record<string, any>` to be a little more specific.

```suggestion
  onComplete?: (rows: Record<string, any>[]) => void // called when all requested rows and columns are parsed
```

Reply:
indeed, as expected here:

https://github.com/hyparam/hyparquet/blob/3c5dbab6496bc38f68c7f2d2463737f2c19ff019/src/hyparquet.js#L14

Reply:
Done